torch/signal/windows/windows.py

```diff
@@ -0,0 +1,297 @@
+import torch
+from math import sqrt
+
+from torch import Tensor
+from torch._prims_common import is_float_dtype
+from torch._torch_docs import factory_common_args, parse_kwargs, merge_dicts
+
+__all__ = [
+    'cosine',
+    'exponential',
+    'gaussian',
+]
+
+window_common_args = merge_dicts(
+    parse_kwargs(
+        """
+    M (int): the length of the window.
+        In other words, the number of points of the exponential window.
+    sym (bool, optional): If `False`, returns a periodic window suitable for use in spectral analysis.
+        If `True`, returns a symmetric window suitable for use in filter design. Default: `True`.
+"""
+    ),
+    factory_common_args,
+    {"normalization": "The window is normalized to 1 (maximum value is 1). However, the 1 doesn't appear if "
+                      "`M` is even and `sym` is `True`."}
+)
+
+
+def _add_docstr(*args):
+    r"""Adds docstrings to a given decorated function.
+
+    Specially useful when then docstrings needs string interpolation, e.g., with
+    str.format().
+    REMARK: Do not use this function if the docstring doesn't need string
+    interpolation, just write a conventional docstring.
+
+    Args:
+        args (str):
+    """
+
+    def decorator(o):
+        o.__doc__ = "".join(args)
+        return o
+
+    return decorator
+
+
+def _window_function_checks(function_name: str, M: int, dtype: torch.dtype, layout: torch.layout) -> None:
+    r"""Performs common checks for all the defined windows.
+     This function should be called before computing any window
+
+     Args:
+         function_name (str): name of the window function.
+         M (int): length of the window.
+         dtype (:class:`torch.dtype`): the desired data type of the window tensor.
```

Comment:
This is consistent with a lot of our doc language, but our docs are not always great, unfortunately.

So question: when `dtype` is set is the window created in that dtype? If so, maybe "the dtype of the constructed tensor" would be clearer and more consistent language?

Same with layout? "the layout of the constructed tensor"?

Reply:
Sure, you know this affects only the `_window_function_checks` function, right?

In `_torch_docs.py` it states the following (which is used for public documentation):

https://github.com/pytorch/pytorch/blob/2aa981ab74df71c8d019f12032ce75910601b52c/torch/_torch_docs.py#L95-L97

Would you like to use the one in `_torch_docs.py`? Or should we stick with your suggestion? I'm good with either option.

Reply:
Using the `_torch_docs.py` default is fine (even though I don't like it, this PR doesn't need to fix all the documentation)

Reply:
Maybe we can create a specific PR for that later on.